### PR TITLE
check for lastBuffer for redraw

### PR DESCRIPTION
**Pull request by Newbrict**
this fixes reusing a texture like a tilemap atlas and changing offsets

### CODE REVIEW

File: shaders.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type defaultShader struct {
 	projY float32
 
 	lastTexture *gl.Texture
+	lastBuffer *gl.Buffer
```
Comment (EtienneBruines) [inline shaders.go]: `gofmt`

File: shaders.go [hunk 2]
```diff
@@ -127,7 +128,7 @@ func (s *defaultShader) Pre() {
 }
 
 func (s *defaultShader) Draw(texture *gl.Texture, buffer *gl.Buffer, x, y, rotation float32) {
-	if s.lastTexture != texture {
+	if s.lastTexture != texture || s.lastBuffer != buffer{
```
Comment (EtienneBruines) [inline shaders.go]: This goes into a separate `if`statement. 

```go
if s.lastBuffer != buffer {
    Gl.BindBuffer(Gl.ARRAY_BUFFER, buffer)
}
``` 

And then that line can be removed this `if` statement.

**Comment by Newbrict** [comment]
don't merge

**Comment by EtienneBruines** [comment]
LGTM. Thank you for your awesome work :smile: 

Will merge when travis is OK with it.

**Pull request merged**